File: rb/features.rb
# BluefinShieldconexMgmt SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module BluefinShieldconexMgmtFeatures
  def self.make_feature(name)
    case name
    when "base"
      BluefinShieldconexMgmtBaseFeature.new
    when "test"
      BluefinShieldconexMgmtTestFeature.new
    else
      BluefinShieldconexMgmtBaseFeature.new
    end
  end
end

File: py/features.py
# BluefinShieldconexMgmt SDK feature factory

from feature.base_feature import BluefinShieldconexMgmtBaseFeature
from feature.test_feature import BluefinShieldconexMgmtTestFeature


def _make_feature(name):
    features = {
        "base": lambda: BluefinShieldconexMgmtBaseFeature(),
        "test": lambda: BluefinShieldconexMgmtTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
